src/jit/lower.cpp

```diff
@@ -240,6 +240,22 @@ GenTree* Lowering::LowerNode(GenTree* node)
 
         case GT_STORE_BLK:
         case GT_STORE_OBJ:
+            if (node->AsBlk()->gtBlkSize == 0)
+            {
+                // Storing a block of zero size is a no-op.
+                GenTree*           nextNode = node->gtNext;
+                bool               isClosed;
+                LIR::ReadOnlyRange blockStoreRange = BlockRange().GetTreeRange(node, &isClosed);
```

Comment:
Hmm, so this would remove both the store and the load. Wouldn't this also remove potential side-effects from the load address tree?

Reply:
Interesting. Removing both the store and the load is fine, I believe, because neither will actually be performed. Regarding other side-effects, that was what I intended the `isClosed` to capture - it would be `false` if there were intervening instructions, e.g. in what used to be a comma in pre-LIR. However, it didn't occur to me that there might be other side-effects, e.g. in computing the addresses. I think that, for this PR, to be conservative, I will file a new issue about removing the zero-length block stores, and remove this part of the change.